### ISSUE TITLE
"Go to Impelmentaion" can not find what types implement the interface

### ISSUE THREAD

**Issue by rockmenjack** [Microsoft/vscode-go#1545]
When trying to find who implement the interface, nothing was found, and an error is printed in debug console:
```
_/home/rockmen1/playground/go/guru_test/... -json implements /home/rockmen1/playground/go/guru_test/src/test_implementation/ifce.go:#30
guru: found packages my_interface (ifce.go) and my_implementation (impl.go) in /home/rockmen1/playground/go/guru_test/src/test_implementation
: Error: Command failed: /home/rockmen1/Tools/bin/guru -scope _/home/rockmen1/playground/go/guru_test/... -json implements /home/rockmen1/playground/go/guru_test/src/test_implementation/ifce.go:#30
guru: found packages my_interface (ifce.go) and my_implementation (impl.go) in /home/rockmen1/playground/go/guru_test/src/test_implementation
```

Steps to Reproduce:
1. Here is a simple step to reproduce, create a simple go code layout as below:
vscode_test
└── src
&nbsp;&nbsp;&nbsp;&nbsp;└── test_implementation
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;└── test.go

2. In the test.go file:
```
package my_interface
type foo interface {
	bar()
}
type impl int
func (i impl) bar() {
}
```
3. Now try to find what implement foo interface, nothing will be found

**Comment by rockmenjack** [comment]
I think "-scope" shall not be used in finding interface implementaions, when -scope is removed, guru is able to give out the correct result.

**Comment by ramya-rao-a** [comment]
I used your code. The result correctly points to the `impl` type

![peek](https://user-images.githubusercontent.com/16890566/36770466-3d0aa0ae-1bff-11e8-8993-7e00cc7c55de.gif)

**Comment by rockmenjack** [comment]
I used these two custom settings:
```
    "go.inferGopath": true,
    "go.toolsGopath": "~/Tools",
```

**Comment by ramya-rao-a** [comment]
The fact that in your logs the file paths are preceded by `_` tells me that this a GOPATH issue

I can repro something similar if I open the GOPATH directly in VS Code.
In your case, GOPATH seems to be `/home/rockmen1/playground/go/guru_test`
The file seems to be in `/home/rockmen1/playground/go/guru_test/src/test_implementation/ifce.go`
What is the folder that you have opened in VS Code? 
Can you open `/home/rockmen1/playground/go/guru_test/src/test_implementation` and try again?

**Comment by rockmenjack** [comment]
> What is the folder that you have opened in VS Code?
It was opened at /home/rockmen1/playground/go/guru_test/


> Can you open /home/rockmen1/playground/go/guru_test/src/test_implementation and try again?

It works in this case. But src/test_implementation is not actually a formal GOPATH.

**Comment by kron4eg** [comment]
@rockmenjack this very much looks like #1536 (now fixed in master)

**Comment by ramya-rao-a** [comment]
@kron4eg This is not the same as #1536. This only occurs when GOPATH is directly opened in VS Code. The way we calculate `scope` gets messed up in this case.

**Comment by ramya-rao-a** [comment]
Fixed. Need some help testing though. All you need to do is
- Download https://github.com/Microsoft/vscode-go/blob/master/Go-latest.vsix
- Run `code --install-extension Go-latest.vsix` 
- If the above fails with `Error: end of central directory record signature not found`, then clone this repo and use the `Go-latest.vsix` file from the cloned repo
- Reload VS Code

**Comment by rockmenjack** [comment]
Tried with the provided extension, it worked!

**Comment by ramya-rao-a** [comment]
This fix is now available in the latest update (0.6.78) to the Go extension
